File: sem1_cs/AnmolSingh_225HSBB004/Assignment13_03_12_22/python/q1.py
#!/usr/bin/python3
#Write a program to count the number of words in a file

fp = open("q1.txt","w")
print("Enter 5 lines into the file:")
for i in range(5):
    line=input()
    fp.write(line+'\n')
fp.close()


fp = open("q1.txt","r")

file=fp.read()
ch=''
ch2=''
count=0
for char in file:
    if ((char!=' ' or char !='\n') and (ch2==' 'or ch2=='\n') and (ch!=' ' or ch !='\n')):
        count+=1
    ch=ch2 
    ch2=char
fp.close()
print("Number of words in the file are:",count)

File: sem1_cs/AnmolSingh_225HSBB004/Assignment13_03_12_22/c/q1.c
/*Write a program to count the number of words in a file*/

#include<stdio.h>

int main(){
	FILE *fptr;
	char ch, ch2, ch3;
	int cntr;
	fptr=fopen("1.txt", "r");
	cntr=0;
	ch="";
	ch2="";
	while(!feof(fptr))
	{
		fscanf(fptr,"%c",&ch3);
		
		if((ch3!=' ' || ch3 !='\n') && (ch2==' '||ch2=='\n') &&(ch!=' ' || ch !='\n') ){
		cntr++;
		}
		ch=ch2; ch2=ch3;
	}

	printf("The number of words in the file is %d.\n", cntr+1);

	return (0);
}
